src/lib/utils/time.ts

```diff
@@ -0,0 +1,92 @@
+const millisecondsPerSecond = 1000
+const secondsPerMinute = 60
+const minutesPerHour = 60
+const hoursPerDay = 24
+const daysPerWeek = 7
+
+const millisecondsPerMinute = millisecondsPerSecond * secondsPerMinute
+const millisecondsPerHour = millisecondsPerMinute * minutesPerHour
+const millisecondsPerDay = millisecondsPerHour * hoursPerDay
+const millisecondsPerWeek = millisecondsPerDay * daysPerWeek
+const previousSevenDays = millisecondsPerWeek
+const lastThirtyDays = millisecondsPerDay * 30
+
+const currentYear = new Date().getFullYear()
+
+const millisecondsSince = (date: Date) => {
+  const now = new Date()
+  return date > now ? 0 : now.getTime() - date.getTime()
+}
+
+export const timeSince = (date: Date) => {
+  const timeDifference = millisecondsSince(date)
+  if (timeDifference === 0) {
+    return '0s'
+  }
+  if (timeDifference < millisecondsPerMinute) {
+    return `${Math.floor(timeDifference / millisecondsPerSecond)}s`
+  } else if (timeDifference < millisecondsPerHour) {
+    return `${Math.floor(timeDifference / millisecondsPerMinute)}m`
+  } else if (timeDifference < millisecondsPerDay) {
+    return `${Math.floor(timeDifference / millisecondsPerHour)}h`
+  } else if (timeDifference < millisecondsPerWeek) {
+    return `${Math.floor(timeDifference / millisecondsPerDay)}d`
+  } else {
+    return `${Math.floor(timeDifference / millisecondsPerWeek)}w`
+  }
+}
+
+export const categorizeDate = (date: Date): string | number => {
+  const now = new Date()
+  if (isToday(date)) {
+    return 'today'
+  } else if (isYesterday(date)) {
+    return 'yesterday'
+  } else if (
+    millisecondsPerDay < millisecondsSince(date) &&
+    millisecondsSince(date) <= previousSevenDays
+  ) {
+    return 'previousSevenDays'
+  } else if (
+    previousSevenDays < millisecondsSince(date) &&
+    millisecondsSince(date) <= lastThirtyDays
+  ) {
+    return 'lastMonth'
+  } else if (date.getFullYear() >= currentYear) {
+    const year = date.getFullYear()
+    return year
+  } else if (now > date) {
+    return 'futureDate'
+  } else {
+    return ''
```

Comment:
Added a check for a future date and an else for any other case because ts couldn't guarantee that the current conditions caught all cases.
Is it better to just have one else that returns empty string instead of dividing it like this?